ci/provisioning/post_provision_config_nodes_EL_7.sh

```diff
@@ -85,36 +117,20 @@ post_provision_config_nodes() {
         done
     fi
     if [ -n "$INST_RPMS" ]; then
-        # shellcheck disable=SC2086
-        dnf -y erase $INST_RPMS
+        dnf -y erase "${INST_RPMS[@]}"
```

Comment:
> or should I revert the change and keep the disable=SC2086

No, I think you are going in the right direction, you just need to convert the string to an array.

> IFS=', ' read -r -a inst_rpms <<< "$INST_RPMS"

Close.  The list is space separated, so setting `$IFS` isn't necessary.  I.e.:

```
$ INST_RPMS="a b c d"; read -r -a inst_rpms <<< "$INST_RPMS"; echo ${#inst_rpms[@]}
4
$ (INST_RPMS=""; read -r -a inst_rpms <<< "$INST_RPMS"; echo ${#inst_rpms[@]})
0
```

But this conversion from string/list to array should go at the very top of this file (right after where the other variables are set) as such:

```
read -r -a inst_rpms <<< "$INST_RPMS"
```

and then the rest of the file should use the `inst_rpms` array instead of `INST_RPMS` by using (for example):

```
if [ $[#inst_rpms[@]} -gt 0 ]; then
    dnf -y erase "${inst_rpms[@]}"
fi
```

Reply:
fixed